### ISSUE TITLE
Error when writing time series when running metrics example

### ISSUE THREAD

**Issue by mattwelke** [GoogleCloudPlatform/opentelemetry-operations-java#139]
I'm running the metrics example locally with the command `./gradlew examples-metrics:run` from the root of this repo. Before I ran this command, I made sure I had a service account provisioned for testing metrics (gave it Monitoring Admin) and I have the `GOOGLE_APPLICATION_CREDENTIALS` env var set to the path on disk to a JSON key file for that service account.

I can see the metric in GCP Monitoring:

![image](https://user-images.githubusercontent.com/7719209/152659333-28346eb9-165f-4e3a-a7d2-111893ddcf88.png)

But I see errors in the logs when it tries to write the time series data points:

```
Running example use case: #51
Feb 05, 2022 4:17:28 PM io.opentelemetry.sdk.metrics.export.PeriodicMetricReader$Scheduled doRun
WARNING: Exporter threw an Exception
com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: One or more TimeSeries could not be written: The set of resource labels is incomplete. Missing labels: (namespace task_id).: timeSeries[0]
        at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:49)
        at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:72)
        at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:60)
        at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:97)
        at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:68)
        at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1074)
        at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
        at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:1213)
        at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:983)
        at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:771)
        at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:563)
        at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:533)
        at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:553)
        at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:68)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:739)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:718)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
        Suppressed: com.google.api.gax.rpc.AsyncTaskException: Asynchronous task failed
                at com.google.api.gax.rpc.ApiExceptions.callAndTranslateApiException(ApiExceptions.java:57)
                at com.google.api.gax.rpc.UnaryCallable.call(UnaryCallable.java:112)
                at com.google.cloud.monitoring.v3.MetricServiceClient.createTimeSeries(MetricServiceClient.java:1526)
                at com.google.cloud.monitoring.v3.MetricServiceClient.createTimeSeries(MetricServiceClient.java:1469)
                at com.google.cloud.opentelemetry.metric.CloudMetricClientImpl.createTimeSeries(CloudMetricClientImpl.java:39)
                at com.google.cloud.opentelemetry.metric.MetricExporter.createTimeSeriesBatch(MetricExporter.java:210)
                at com.google.cloud.opentelemetry.metric.MetricExporter.export(MetricExporter.java:195)
                at io.opentelemetry.sdk.metrics.export.PeriodicMetricReader$Scheduled.doRun(PeriodicMetricReader.java:125)
                at io.opentelemetry.sdk.metrics.export.PeriodicMetricReader$Scheduled.run(PeriodicMetricReader.java:117)
                at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
                at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
                at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
                ... 3 more
Caused by: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: One or more TimeSeries could not be written: The set of resource labels is incomplete. Missing labels: (namespace task_id).: timeSeries[0]
        at io.grpc.Status.asRuntimeException(Status.java:535)
        ... 10 more
```

And I don't see any data in GCP Monitoring when I select the metric:

![image](https://user-images.githubusercontent.com/7719209/152659399-afad2305-fd8f-4d6c-ae56-92cafa4f8516.png)

I'm not sure how to interpret that error. One way I can interpret it is that it's telling me there are two labels missing called "namespace" and "task_id". But I don't understand why those labels exist in the first place. I don't see any code in `MetricsExporterExample.java` that involves creating the metric descriptor. I don't see any code that mentions "namespace" or "task_id".

I'm stuck at this point because I'm evaluating whether we can switch from OpenCensus to OpenTelemetry for our apps on GCP. I was hoping to start with a working minimal example and then poke around in the code to see what makes it work.

**Comment by aabmass** [comment]
Thanks for the bug report @mattwelke. This was caused by the change in #134. The `namespace` and `task_id` labels it's complaining about are not metric labels but Monitored Resource labels for type `generic_task`.

The example would pass if you ran it on GCE or GKE where resource detectors can create a valid Monitored Resource. But I think there is a bug, as it should work on your local machine with credentials as well. I'll take a look

**Comment by mattwelke** [comment]
@aabmass Thanks for taking a look. To give you a bit more backstory, we started using Stackdriver and happened to prefer the OpenCensus API instead of sending data points to GCP directly. We like the fact that we don't have to deploy our apps to GCE or GKE to use it too. If we switch to OpenTelemetry, it'd be nice if we continue to have that freedom. Can you clarify... is that the goal of this exporter, that one could send metrics to GCP even if they're not running their application on GCE, GKE, or even on GCP at all?

**Comment by aabmass** [comment]
> is that the goal of this exporter, that one could send metrics to GCP even if they're not running their application on GCE, GKE, or even on GCP at all?

Yes, this exporter will work regardless of where you run your application. #140 fixes the exporter so the example will run as you expected.

For a real application not running on GCP infra, you should set the [OpenTelemetry Resource](https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-sdk-metrics/latest/io/opentelemetry/sdk/metrics/SdkMeterProviderBuilder.html) attributes `service.name` (name of the application/service), `service.instance.id` (unique ID for this instance of your app/service) and `service.namespace` (the namespace your service belongs to). The exporter will use those to set the `generic_task` [monitored resource](https://cloud.google.com/monitoring/api/v3/metric-model#resource-overview) labels. This will help avoid writing overlapping metrics and help with quota as you're allowed [200,000 active time series per single monitored resource "instance"](https://cloud.google.com/monitoring/quotas#:~:text=Active%20time%20series%20from%20workload%20metrics%20per%20monitored%20resource5).

Let me know if you have any more questions 🙂

**Comment by aabmass** [comment]
Yes those docs are relevant. 

> They describe it as optional, but I'm assuming that's at the specification level, and that if you're using a particular exporter that requires it, you'd add it. I'm assuming the GCP exporter is an example of such an exporter.

The GCP exporter doesn't require it necessarily, we can use an empty string if it's missing (fixed in #140).
